plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -703,9 +705,24 @@ function OPDSBrowser:showDownloads(item)
     })
     table.insert(buttons, {
         {
-            text = _("Cancel"),
+            text = _("Book cover"),
+            enabled = item.image ~= nil or item.thumbnail ~= nil,
             callback = function()
-                UIManager:close(self.download_dialog)
+                local cover_link = item.image or item.thumbnail
+                local function file_downloaded_callback(cover_file)
+                    local function image_viewer_close_callback()
+                        os.remove(cover_file)
+                    end
+                    local ImageViewer = require("ui/widget/imageviewer")
+                    UIManager:show(ImageViewer:new{
+                        file = cover_file,
+                        fullscreen = true,
+                        with_title_bar = false,
+                        file_do_cache = false,
+                        close_callback = image_viewer_close_callback,
+                    })
+                end
+                self:downloadFile("tmp_cover.jpg", cover_link, file_downloaded_callback)
```

Comment:
> * you save it to a local file, because that's what this OPDS plugin downloadFile() - it's a bit ugly

Do you see other ways?
The cover on opds server is a file. ImageViewer input is a bb image of a file.

Reply:
> Currently for ImageViewer file_do_cache = true

Yes, that ImageWidget default is good, as when ImageWidget is provided with a file=, it's usually for icons, so we want that cache.
But for ImageViewer, I think we don't want that. Haven't looked if there are currently usage where we provide a file - I think no, and if you are the first one, you may set a default of `file_do_cache = false` in ImageViewer.

> Do you see other ways?

I'm not saying you should change it :)
But usually, it's a matter of playing with the `sink=`  of the http/socket stuff, see `local function getUrlContent` in frontend/ui/wikipedia.lua, which I think is also used to fetch Wikipedia image for display.

Reply:
I see, I'll try renderImageData.